commit 70e99e76b5f63265f6b2700f8af77ece4f1dd11b
Author: Kefu Chai <kchai@redhat.com>
Date:   Wed Nov 25 18:44:04 2020 +0800

    mgr: do not migrate conf from config-key store to new-style conf
    
    since all module options are using the new-style config framework.
    the migration is offered for the use case of upgrade from luminous to mimic,
    since pacific can only be upgraded from octopus. the mimic monitors are alreay
    able to populate the configurations to mgr, not to mention the octopus
    monitors, so there is no need to migrate the options stored in config-key
    store anymore.
    
    Signed-off-by: Kefu Chai <kchai@redhat.com>

diff --git a/src/mgr/PyModuleRegistry.cc b/src/mgr/PyModuleRegistry.cc
index 2aa556cc9d5..ad461aa47bc 100644
--- a/src/mgr/PyModuleRegistry.cc
+++ b/src/mgr/PyModuleRegistry.cc
@@ -445,83 +445,3 @@ void PyModuleRegistry::handle_config_notify()
     active_modules->config_notify();
   }
 }
-
-void PyModuleRegistry::upgrade_config(
-    MonClient *monc,
-    const std::map<std::string, std::string> &old_config)
-{
-  // Only bother doing anything if we didn't already have
-  // some new-style config.
-  if (module_config.config.empty()) {
-    dout(1) << "Upgrading module configuration for Mimic" << dendl;
-    // Upgrade luminous->mimic: migrate config-key configuration
-    // into main configuration store
-    for (auto &i : old_config) {
-      auto last_slash = i.first.rfind('/');
-      const std::string module_name = i.first.substr(4, i.first.substr(4).find('/'));
-      const std::string key = i.first.substr(last_slash + 1);
-
-      const auto &value = i.second;
-
-      // Heuristic to skip things that look more like stores
-      // than configs.
-      bool is_config = true;
-      for (const auto &c : value) {
-        if (c == '\n' || c == '\r' || c < 0x20) {
-          is_config = false;
-          break;
-        }
-      }
-
-      if (value.size() > 256) {
-        is_config = false;
-      }
-
-      if (!is_config) {
-        dout(1) << "Not migrating config module:key "
-                << module_name << " : " << key << dendl;
-        continue;
-      }
-
-      // Check that the named module exists
-      auto module_iter = modules.find(module_name);
-      if (module_iter == modules.end()) {
-        dout(1) << "KV store contains data for unknown module '"
-                << module_name << "'" << dendl;
-        continue;
-      }
-      PyModuleRef module = module_iter->second;
-
-      // Parse option name out of key
-      std::string option_name;
-      auto slash_loc = key.find("/");
-      if (slash_loc != std::string::npos) {
-        if (key.size() > slash_loc + 1) {
-          // Localized option
-          option_name = key.substr(slash_loc + 1);
-        } else {
-          // Trailing slash: garbage.
-          derr << "Invalid mgr store key: '" << key << "'" << dendl;
-          continue;
-        }
-      } else {
-        option_name = key;
-      }
-
-      // Consult module schema to see if this is really
-      // a configuration value
-      if (!option_name.empty() && module->is_option(option_name)) {
-        module_config.set_config(monc, module_name, key, i.second);
-        dout(4) << "Rewrote configuration module:key "
-                << module_name << ":" << key << dendl;
-      } else {
-        dout(4) << "Leaving store module:key " << module_name
-                << ":" << key << " in store, not config" << dendl;
-      }
-    }
-  } else {
-    dout(10) << "Module configuration contains "
-             << module_config.config.size() << " keys" << dendl;
-  }
-}
-
